### PR TITLE
Task collection subclient

### PR DESCRIPTION
**Pull request by mhamas**

### CODE REVIEW

File: src/apify_client/clients/resource_clients/task_collection.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+from ..._utils import _filter_out_none_values_recursively
+from ..base.resource_collection_client import ResourceCollectionClient
+
+
+class TaskCollectionClient(ResourceCollectionClient):
+    """Sub-client for manipulating tasks."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the TaskCollectionClient with the passed arguments."""
```
Comment (Equidem) [inline src/apify_client/clients/resource_clients/task_collection.py:11]: Why is there suddenly the addition of "with the passed arguments"?

File: src/apify_client/clients/resource_clients/task_collection.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        super().__init__(*args, resource_path='actor-tasks', **kwargs)
+
+    def list(self, *, limit: Optional[int] = None, offset: Optional[int] = None, desc: Optional[bool] = None) -> Dict:
+        """List the available tasks.
+
+        https://docs.apify.com/api/v2#/reference/actor-tasks/task-collection/get-list-of-tasks
+
+        Args:
+            limit: How many tasks to retrieve
```
Comment (Equidem) [inline src/apify_client/clients/resource_clients/task_collection.py:20]: "How many tasks to list"
(for the sake of consistency)

File: src/apify_client/clients/resource_clients/task_collection.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            offset: What tasks to include as first when retrieving the list
```
Comment (Equidem) [inline src/apify_client/clients/resource_clients/task_collection.py:21]: "What task to include ..."

File: src/apify_client/clients/resource_clients/task_collection.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        task_input: Optional[Dict] = {},
+    ) -> Dict:
+        """Create a new task.
+
+        https://docs.apify.com/api/v2#/reference/actor-tasks/task-collection/create-task
+
+        Args:
+            actor_id (string): Id of the actor that should be run
+            name (string): Name of the task
+            build (str, optional): Specifies the actor build to run. It can be either a build tag or build number.
```
Comment (Equidem) [inline src/apify_client/clients/resource_clients/task_collection.py:46]: Same gripe as in the TaskClient ;)

File: src/apify_client/clients/resource_clients/task_collection.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                                   By default, the run uses the build specified in the task settings (typically latest).
+            memory_mb (int, optional): Memory limit for the run, in megabytes. By default, the run uses a memory limit specified in the task settings.
```
Comment (Equidem) [inline src/apify_client/clients/resource_clients/task_collection.py:48]: memory_mbytes? (as in the TaskClient)

**Review by Equidem — CHANGES_REQUESTED** [review]
Again, just naming and commenting nitpicks.

File: src/apify_client/clients/resource_clients/task_collection.py (hunk 5 above)
Comment (mhamas) [reply src/apify_client/clients/resource_clients/task_collection.py:48]: I think that from `_mb` suffix it should be sufficiently clear that it means `megabytes`, isn't it?

Comment (mnmkng) [reply src/apify_client/clients/resource_clients/task_collection.py:48]: It should be consistent with the API though. And API uses `mbytes`. We already had different names between API and client v0. It was not working well at all.

**Comment by mnmkng** [comment]
A "mention" notification popped for me, but I can't find any. Did you need anything from me?

**Comment by mhamas** [comment]
@mnmkng might have been from here: https://github.com/apify/apify-client-python/pull/32/files#r576008928?

File: src/apify_client/clients/resource_clients/task_collection.py (hunk 5 above)
Comment (mhamas) [reply src/apify_client/clients/resource_clients/task_collection.py]: Ok, I renamed to `mbytes`

**Review by Equidem — APPROVED** [review]
Approved once you deal with the linting issue :)

**Review by fnesveda — CHANGES_REQUESTED** [review]

File: src/apify_client/clients/resource_clients/task_collection.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+    def create(
+        self,
+        *,
+        actor_id: str,
+        name: str,
+        build: Optional[str] = None,
+        timeout_secs: Optional[int] = None,
+        memory_mbytes: Optional[int] = None,
+        task_input: Optional[Dict] = {},
```
[the remark this answers is not in the document]
Comment (mhamas) [reply src/apify_client/clients/resource_clients/task_collection.py]: addressed

**Review by fnesveda — APPROVED** [review]

**Pull request merged**